src/graph/entities/edge.c

```diff
@@ -17,15 +17,29 @@ NodeID Edge_GetSrcNodeID
 	const Edge *edge
 ) {
 	ASSERT(edge);
-	return edge->srcNodeID;
+	if(edge->src_id == INVALID_ENTITY_ID) {
+		return INVALID_ENTITY_ID;
+	}
+	if(edge->src_id & MSB_MASK) {
+		NodeID *id = (NodeID *)(CLEAR_MSB(edge->src_id));
+		return *id;
+	}
+	return edge->src_id;
 }
 
 NodeID Edge_GetDestNodeID
 (
 	const Edge *edge
 ) {
 	ASSERT(edge);
-	return edge->destNodeID;
+	if(edge->dest_id == INVALID_ENTITY_ID) {
+		return INVALID_ENTITY_ID;
+	}
+	if(edge->dest_id & MSB_MASK) {
+		NodeID *id = (NodeID *)(CLEAR_MSB(edge->dest_id));
```

Comment:
Why do we need to deal with `CLEAR_MSB` at this point ?
I'm expecting `Edge_SetSrcNode` receive and store a "valid" node ID. MSB cleared.

Reply:
to avoid complexity consider switching to `ReserveNodeID` to guarantee intermediate nodes are associated with an ID.

Reply:
done